### PR TITLE
Attempt to start

### PR DESCRIPTION
**Pull request by deads2k**
This is an attempt using glide.  I've managed to generate building code, so that's nice.

I used glide to handle the vendor dir, but I'm unsure how repeatable it really is.

The clients are under 
`<group>/{internalclientset,clientset,informers,listers}`

@liggitt @sttts @smarterclayton let's argue about:
 1.  whether we want to support `go get`-ability.  I don't, vendor or you aren't repeatable and if you aren't repeatable, then I don't feel too bad about weird things happening
 2. whether we want to publish internal clients.  I think internal clients could be useful for an orderly transition of our origin projects that could use this, but I wouldn't want to support them as stable.
 3. location of the generated code.

### CODE REVIEW

**Comment by smarterclayton** [comment]
I think we want to support go get to a degree

**Comment by smarterclayton** [comment]
In the sense that there's a reason people want to do go get.

I'm less worried about internal clients - I think evidence from upstream is that only a small subset of people needed them.

**Comment by sttts** [comment]
About 1: I don't care much about go-get'ability. But I care about integrating this into other projects. The procedure for kube/client-go (compare https://github.com/kubernetes/client-go/issues/259#issuecomment-320320158) is
- do checkout client-go, 
- then godep restore from there,
- go to your project and godep save.

Does this work with glide? https://github.com/Masterminds/glide/issues/366 suggests it doesn't.

File: glide.lock [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  version: d6023ce2651d8eafb5c75bb0c7167536102ec9f5
+- name: github.com/fsouza/go-dockerclient
+  version: 3f9370a4738ba8d0ed6eea63582ca6db5eb48032
+  subpackages:
+  - external/github.com/Sirupsen/logrus
+  - external/github.com/docker/docker/opts
+  - external/github.com/docker/docker/pkg/archive
+  - external/github.com/docker/docker/pkg/fileutils
+  - external/github.com/docker/docker/pkg/homedir
+  - external/github.com/docker/docker/pkg/idtools
```
Comment (sttts) [inline glide.lock]: omg, we vendored the whole universe.

File: hack/update-generated-clientset.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
```
Comment (sttts) [inline hack/update-generated-clientset.sh]: I am not a fan of these 50 liner bash scripts in every projects right now. But it's the best we have right now :-/

Comment (sttts) [inline hack/update-generated-clientset.sh]: Can we call it `update-codegen.sh` at least?

Comment (sttts) [inline hack/update-generated-clientset.sh]: `#!/bin/bash -e` for proper error handling.

**Comment by deads2k** [comment]
How about this as a starting point and after it merges I enable travis and we go from there?

**Comment by sttts** [comment]
As long as we leave the "Not ready yet" in the readme, lgtm.

**Comment by deads2k** [comment]
> As long as we leave the "Not ready yet" in the readme, lgtm.

Definitely will.  Merging this to build on .

**Comment by mfojtik** [comment]
👍

File: glide.lock (hunk 1 above)
Comment (mfojtik) [reply glide.lock]: we should find why we need those  and make issues to improve our code hygiene for those in corresponding teams :)

**Comment by ralphg6** [comment]
:+1:

**Pull request merged**